src/app/checkout/step-2/existingPaymentMethods/existingPaymentMethods.component.spec.js

```diff
@@ -329,6 +343,49 @@ describe('checkout', () => {
           expect(self.controller.onPaymentChange).toHaveBeenCalledWith({ selectedPaymentMethod: undefined })
           expect(self.controller.orderService.storeCoverFeeDecision).not.toHaveBeenCalled()
         })
+
+        it('should reset securityCode viewValue on switch payment', () => {
+          self.controller.creditCardPaymentForm.securityCode.$viewValue = '123'
+          self.controller.selectedPaymentMethod = { 'card-type': 'Visa', self: { type: 'cru.creditcards.named-credit-card', uri: 'selected uri' }, selectAction: 'some uri' }
+          self.controller.switchPayment()
+
+          expect(self.controller.creditCardPaymentForm.securityCode.$setViewValue).toHaveBeenCalledWith('')
+          expect(self.controller.creditCardPaymentForm.securityCode.$render).toHaveBeenCalled()
+        })
+      })
+
+      describe('addCvvValidators', () => {
+        it('should add validator functions to creditCardPaymentForm.securityCode', () => {
+          self.controller.addCvvValidators()
```

Comment:
Since this function is just adding a watch on the scope, the way this test is built feels a bit flaky. You aren't changing `$ctrl.creditCardPaymentForm.securityCode.$viewValue` as part of this test, which means you are relying on some other test in this file, or possibly the initialization of the `creditCardPaymentForm` to be doing it for you, which may run earlier or later than this test. This test should be fully self-contained. Maybe we need to reset it somehow first then change the value then expect all within this test.